### PR TITLE
refactor(app): [ESP-3382] initialisation setup

### PR DESCRIPTION
**Pull request by henryhobhouse**
JIRA Ticket: https://trayio.atlassian.net/browse/ESP-3382?atlOrigin=eyJpIjoiZGNhMDdkODI1MWNjNGIyNDk0Njc5OTg0YTcwYzQ5M2MiLCJwIjoiaiJ9

Shotgun ticket to improve how to setup the sample app as the enterprise squad are likely going to use this to debug the issue with workspaces and I felt that it wasn't that accessible in its present form. 

* Update app to use `.env` files to retrieve required environmental configurations. This has a principle purpose that you don't have to add in the same information into the setup script every time your run the app. 
* Improve the readme to reflect the above change in addition to giving direction on how to retrieve the partner token which is not readily obvious to non embedded devs.

### CODE REVIEW

**Comment by henryhobhouse** [comment]
@floroz  You might be the perfect reviewer on this as I don't think you have had to setup up this app yet?

File: README.md [hunk 1]
```diff
@@ -14,59 +14,89 @@ In this repo is a sample webapp which runs on top of the Tray.io Embedded Editio
...
+
+ You can retrieve the token for any environment by visiting the Tray app instance for that environment i.e. 
+ 
+ prod -> https://app.tray.io
+ stg -> https://app.staging.tray.io
+ fe-stag -> https://app.frontend-staging.tray.io
+
+You will then need to log on as a embedded user and visit `Settings & people` -> `Tokens`
+
+![getting-token](https://github.com/trayio/embedded-edition-sample-app/.git-images/getting-token)
```
Comment (henryhobhouse) [inline README.md]: This image won't show until the PR is merged as assumes the image is on master.

File: README.md [hunk 2]
```diff
@@ -14,59 +14,89 @@ In this repo is a sample webapp which runs on top of the Tray.io Embedded Editio
...
+
+ You can retrieve the token for any environment by visiting the Tray app instance for that environment i.e. 
+ 
+ prod -> https://app.tray.io
+ stg -> https://app.staging.tray.io
+ fe-stag -> https://app.frontend-staging.tray.io
+
+You will then need to log on as a embedded user and visit `Settings & people` -> `Tokens`
+
+![getting-token](https://github.com/trayio/embedded-edition-sample-app/blob/master/.images/getting-token)
```
Comment (henryhobhouse) [inline README.md:68]: This image won't show until the PR is merged as assumes the image is on master.

File: package.json [hunk 3]
```diff
@@ -28,17 +29,19 @@
     "uuid": "^3.2.1"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "api": "nodemon server/server.js --exec babel-node --presets es2015,stage-2",
+    "start": "concurrently \"npm run api\" \"npm run start-client\"",
```
Comment (henryhobhouse) [inline package.json:32]: This will show the logs of both the api and client in the same process

File: package.json [hunk 4]
```diff
@@ -28,17 +29,19 @@
...
+    "start-client": "HTTPS=true react-scripts start",
+    "api": "BABEL_DISABLE_CACHE=1 nodemon --exec babel-node -r dotenv/config server/server.js --presets @babel/preset-env",
```
Comment (henryhobhouse) [inline package.json:34]: Uses dotenv to import the env variables (for tray config)

File: package.json [hunk 5]
```diff
@@ -28,17 +29,19 @@
...
     "eject": "react-scripts eject"
   },
   "devDependencies": {
-    "babel-cli": "^6.26.0",
-    "babel-preset-es2015": "^6.24.1",
-    "babel-preset-stage-2": "^6.24.1",
-    "nodemon": "^1.18.7",
+    "@babel/core": "^7.14.5",
+    "@babel/node": "^7.14.5",
+    "@babel/preset-env": "^7.14.5",
```
Comment (henryhobhouse) [inline package.json:42]: Babel v6 has issues with using dotenv in both determining path to package and also permeating env vars to child modules.

**Comment by floroz** [comment]
@henryhobhouse I'll need some context about this app/repo as I've never seen/used it before

**Review by huxley-tray — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (huxley-tray) [inline package.json:32]: noice

**Review by huxley-tray — APPROVED** [review]

**Pull request merged**